### ISSUE TITLE
Invalid git tags being produced of the form 'v0.0.0-DATE-SHA' 

### ISSUE THREAD

**Issue by 08d2** [osmosis-labs/osmosis#4004]
A recent change apparently began producing git tags on this repository of the form v0.0.0-DATE-SHA.

- https://github.com/osmosis-labs/osmosis/releases/tag/osmomath%2Fv0.0.2-20230103003823-2ac5d3561928
- https://github.com/osmosis-labs/osmosis/releases/tag/v0.0.0-20230111151925-129792310797
- https://github.com/osmosis-labs/osmosis/releases/tag/v0.0.0-20230111152143-f81b9368fb8b
- https://github.com/osmosis-labs/osmosis/releases/tag/v0.0.0-20230112084519-84d6450e1670

This syntax is used by Go modules as a pseudo-version when resolving dependencies. These strings are not intended to be used as literal git tags on the source repository, and can easily result in unpredictable or broken behavior for consumers.

What change started producing these tags, and by what rationale?

**Comment by ValarDragon** [comment]
Yeah this is surprising, thanks for flagging this. This shouldn't have happened, looked like it got pushed yesterday.

(Actually got pushed by someone who should not have perms to push tags)

Do you know of a way we can change git permissions to let people make branches on the repo, but not be allowed to push tags. I think that was the root cause of the issue this time.

**Comment by ValarDragon** [comment]
I am hopeful that getting this tool for hash linking all of our modules during local edits: https://github.com/osmosis-labs/osmosis/pull/3906 - should alleviate these issues.

Seems to be a lot of confusion / manual overhead going on right now with this local linking by hand.

**Comment by peterbourgon** [comment]
> Do you know of a way we can change git permissions to let people make branches on the repo, but not be allowed to push tags. I think that was the root cause of half of it this time.

I'm not aware of a way to do this. Normally these users would not be granted any push permissions and would use their own forks to work on e.g. PRs.

> I am hopeful that getting this tool for hash linking all of our modules during local edits: https://github.com/osmosis-labs/osmosis/pull/3906 - should alleviate these issues.

Is there a document or something that explains the intent of this tool? Based on a superficial reading of the code, I'm not sure that Go modules works the way that this tool expects.
